File: unmastered/ctci_problems/linked_lists/2.7.js
var LinkedList = require('../../../mastered/data_structures/linked_list').LinkedList;
var Node = require('../../../mastered/data_structures/linked_list').Node;

function intersection(list1, list2){
	let map = {};
	let ptr1 = list1.head.next;
	while (ptr1 !== list1.tail){
		map[ptr1] = true;
		ptr1 = ptr1.next;
	}
	let ptr2 = list2.head.next;
	while (ptr2 !== list2.tail){
		if (map[ptr2]){
			ptr2;
		}
		ptr2 = ptr2.next;
	}
	return false;
}

let list1 = new LinkedList();
list1.addNode(1);
list1.addNode(2);
list1.addNode(3);

let list2 = new LinkedList();
list2.addNode(1);
list2.addNode(2);
list2.addNode(3);

if (intersection(list1, list2)){
	throw "ERROR 1"
}

let intersection = new Node(10);
intersection.list1.head.next

File: unmastered/ctci_problems/linked_lists/2.7.py
class Node(object):
	def __init__(self, val):
		self.val = val;
		self.next = None;
		self.prev = None;

class LinkedList(object):
	def __init__(self):
		self.head = Node(None);
		self.tail = Node(None);
		self.head.next = self.tail;
		self.tail.prev = self.head;
		self.head.prev = None;
		self.tail.next = None;

	def addNode(self, node):
		self.tail.prev.next = node;
		node.prev = self.tail.prev;
		self.tail.prev = node;
		node.next = self.tail;

	def prnt(self):
		currPtr = self.head.next
		while (currPtr != self.tail):
			print currPtr.val;
			currPtr = currPtr.next;

"""
	get length A.  Get Length B.  In the process of doing so check if last nodes are equal, if they arnt then just return None
	  Whichever one is bigger, move head ptr down by the difference between 2 lengths.  
	now move each ptr until they are equal, and return it.
"""
def intersection(listA, listB):

